internal/manifest/v1beta1/spec_resolver.go

```diff
@@ -174,27 +173,31 @@ func (m *ManifestSpecResolver) getValuesFromConfig(
 }
 
 var (
-	ErrChartConfigObjectInvalid = errors.New("chart config object of .spec.config is invalid")
-	ErrConfigObjectInvalid      = errors.New(".spec.config is invalid")
+	ErrConfigObjectInvalid = errors.New(".spec.config is invalid")
 )
 
-func parseInstallConfigs(decodedConfig interface{}) ([]interface{}, error) {
+type ConfigsYaml struct {
+	Configs []interface{}
+}
+
+func ParseInstallConfigs(decodedConfig interface{}) ([]interface{}, error) {
 	var configs []interface{}
-	installConfigObj, decodeOk := decodedConfig.(map[string]interface{})
-	if !decodeOk {
+	var configsValues ConfigsYaml
+	configsValuesYamlObj, err := json.Marshal(decodedConfig)
```

Comment:
Do we have a specific reason to use `json.marshal` and `unmarshal`? Suppose you check the code how this `decodedConfig` get generated. In that case, it's actually from a yaml file, which already go through the yaml to json decoding, and be aware, the operation to do those transferring is expansive, will affect the reconcile performance.

```
err := yaml.NewYAMLOrJSONDecoder(blob, internal.YamlDecodeBufferSize).Decode(&decodedConfig)
```
Besides this, I suggest for this case, the test should test directly from reading an empty config.yaml to check the final parsed config []interface{}